cdc/generation.cc

```diff
@@ -693,28 +693,28 @@ void generation_service::on_change(gms::inet_address ep, gms::application_state
         return;
     }
 
-    auto ts = gms::versioned_value::cdc_generation_id_from_string(v.value);
-    cdc_log.debug("Endpoint: {}, CDC generation ID change: {}", ep, ts);
+    auto gen_id = gms::versioned_value::cdc_generation_id_from_string(v.value);
+    cdc_log.debug("Endpoint: {}, CDC generation ID change: {}", ep, gen_id);
 
-    handle_cdc_generation(ts).get();
+    handle_cdc_generation(gen_id).get();
 }
 
 future<> generation_service::check_and_repair_cdc_streams() {
     if (!_joined) {
         throw std::runtime_error("check_and_repair_cdc_streams: node not initialized yet");
     }
 
-    auto latest = _gen_id;
+    std::optional<cdc::generation_id> latest = _gen_id;
```

Comment:
why not auto?

Reply:
Wanted to make it obvious that the latest ID may be missing.